mantidproject/mantid
# Indirect IqtFit - Prevent plotting of one spectra

**Description of work.**
**A.** This PR ensures that you cannot attempt a plot of a workspace that has only one data point to be plotted.
**B.** This PR also makes sure the `Fit Single Spectrum` button is disabled when fitting a single spectrum for instance in ConvFit.

**To test:**
**A.**
1. Go to `Interfaces` > `Indirect` > `Data Analysis`
2. Go to the I(Q, T) tab
3. Load the irs26176_graphite002_red.nxs file from the sample data
4. Load the resolution file irs26173_graphite002_res.nxs from the sample data
5. Click Run, a new workspace with the suffix _iqt should appear

6. Go to the I(Q, T) Fit tab
7. Load the _iqt workspace 
8. Set Exponential to 1
9. Set EndX to 0.14
10. Change `Fit spectra` to String and set this to 3
11. Click Run

`Plot` in the output options should be disabled as there is only one data point to be plotted.

**B.**
1. Go to the ConvFit tab
2. Load the data below
3. Select `TeixeiraWater` as Fit Type
4. Click `Fit Single Spectrum` and make sure it is disabled while fitting and then re-enabled

**Data Files**
[irs26176_graphite002_red.zip](https://github.com/mantidproject/mantid/files/2634132/irs26176_graphite002_red.zip)
[irs26173_graphite002_res.zip](https://github.com/mantidproject/mantid/files/2634133/irs26173_graphite002_res.zip)

Fixes #24151

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
